mpqp/execution/providers/ibm.py

```diff
@@ -137,6 +151,193 @@ def check_job_compatibility(job: Job):
             )
 
 
+@typechecked
+def generate_qiskit_noise_model(
+    circuit: QCircuit,
+) -> tuple["Qiskit_NoiseModel", QCircuit]:
+    """
+    Generate a Qiskit Aer NoiseModel from the list of MPQP NoiseModels attached to the QCircuit in parameter.
+
+    Args:
+        circuit: QCircuit used to generate the Qiskit NoiseModel.
+
+    Returns:
+        Qiskit_NoiseModel: A Qiskit NoiseModel combining the provided noise models.
+
+    Note:
+        The qubit order is reversed to match Qiskit's qubit ordering conventions.
+
+    """
+    from qiskit_aer.noise import NoiseModel as Qiskit_NoiseModel
+
+    noise_model = Qiskit_NoiseModel()
+
+    modified_circuit = deepcopy(circuit)
+
+    used_qubits = set().union(
+        *(
+            inst.connections()
+            for inst in modified_circuit.instructions
+            if isinstance(inst, Gate)
+        )
+    )
+    modified_circuit.instructions.extend(
+        [
+            Id(qubit)
+            for qubit in range(modified_circuit.nb_qubits)
+            if qubit not in used_qubits
+        ]
+    )
+
+    gate_instructions = modified_circuit.get_gates()
+
+    noisy_identity_counter = 0
+
+    for noise in modified_circuit.noises:
+        qiskit_error = noise.to_other_language(Language.QISKIT)
+        if TYPE_CHECKING:
+            from qiskit_aer.noise.errors.quantum_error import QuantumError
+
+            assert isinstance(qiskit_error, QuantumError)
+
+        # If all qubits are affected
+        if len(noise.targets) == modified_circuit.nb_qubits:
+            if len(noise.gates) != 0:
+                for gate in noise.gates:
+                    size = gate.nb_qubits
+                    if TYPE_CHECKING:
+                        assert isinstance(size, int)
+
+                    if isinstance(noise, DimensionalNoiseModel):
+                        if size != noise.dimension:
+                            continue
```

Comment:
of even
```py
if isinstance(noise, DimensionalNoiseModel) and size == noise.dimension:
    ...
```

Reply:
> of even
> 
> ```python
> if isinstance(noise, DimensionalNoiseModel) and size == noise.dimension:
>     ...
> ```

This one pyright may not like

Reply:
> we can change to == and delete continue

The thing is that we want to skip this exact case, all the other cases are fine, but are not express shortly in a `if`. Because they concern non dimensional noise model, and dimensional noise models with right gate size.

So that is why it is better to check the discarding condition, and not the complement. But I agree that in general, using continue may reveal a bad implementation.

If you have better ideas, welcome

Reply:
> This one pyright may not like

Pyright was fine with is but it was no correct, but Julien's initial proposition was, so I implemented it